gsoc/gsoc2016/ideas_sunpy.md

```diff
@@ -0,0 +1,179 @@
+Recently, a new downloader front end (originally named `UnifiedDownloader`, now affectionately known as `Fido`) has been developed, this provides a Factory Class, with which various download clients (such as the VSO) can register with, providing information about which attrs and attr values that client supports. Using this approach, the `Fido` downloader provides a single interface to the many different services SunPy supports.
+The first part of this project will be to update the database module to support the new `Fido` interface, specifically by using `Fido` inside the database to retrieve data.
+
+The second part of the project will be to update the caching mechanism implemented in the database module. The current caching system serialises the users VSO query and stores it as JSON, upon the user requesting another query, the query will be compared to the cache of serialised queries and if a match is found, the results from the cached query returned.
+This mechanism is limiting in that if the user requests 100 records in query A and 100 records in query B, but 50 of the records requested in both queries are the same (i.e. two overlapping time windows) then the 50 records will be re-downloaded as the cache of query A will not match query B.
+The updated caching system will store the records a query returns (before the data is downloaded) and then link the results of a query to the records in the database (once the data has been downloaded). Then when records are retrieved from a web service, any records that are stored in the cache table can be skipped for retrieval from the web service and returned from the records in the database.
+This will allow the caching of partial queries rather than whole queries as is currently implemented.
+
+This project aims to achieve the following things:
+
+1. Update the current implementation of the database using the VSO attributes to use the slightly refactored `Fido` attributes and use `Fido` inside the database to download data from the VSO.
+1. Implement a new caching mechanism bases of the results of Queries with `Fido` rather than the current caching which is based upon the VSO query.
+
+A successful proposal will schedule updates to the database package in small sections, rather than in one large pull request. The work should be understood and broken down into individual sections.
+
+There are various other maintenance tasks which need undertaking (https://github.com/sunpy/sunpy/labels/Database) which would be a good way for someone interested in this project to familiarise themselves with the codebase.
+
+
+
+### Real time data access and visualisation tools
+
+*Suggested Mentor(s):* [David Perez-Suarez](http://github.com/dpshelio), [Jack Ireland](https://github.com/wafels)
```

Comment:
Ok by me - thanks!

On Thu, Feb 18, 2016 at 3:21 PM, dpshelio <notifications@github.com> wrote:

> In gsoc/gsoc2016/ideas_sunpy.md
> <https://github.com/OpenAstronomy/openastronomy.github.io/pull/34#discussion_r53376520>
> :
>
> > +This will allow the caching of partial queries rather than whole queries as is currently implemented.
> > +
> > +This project aims to achieve the following things:
> > +
> > +1. Update the current implementation of the database using the VSO attributes to use the slightly refactored `Fido` attributes and use `Fido` inside the database to download data from the VSO.
> > +1. Implement a new caching mechanism bases of the results of Queries with `Fido` rather than the current caching which is based upon the VSO query.
> > +
> > +A successful proposal will schedule updates to the database package in small sections, rather than in one large pull request. The work should be understood and broken down into individual sections.
> > +
> > +There are various other maintenance tasks which need undertaking (https://github.com/sunpy/sunpy/labels/Database) which would be a good way for someone interested in this project to familiarise themselves with the codebase.
> > +
> > +
> > +
> > +### Real time data access and visualisation tools
> > +
> > +*Suggested Mentor(s):* [David Perez-Suarez](http://github.com/dpshelio), [Jack Ireland](https://github.com/wafels)
>
> @wafels <https://github.com/wafels> I've added you here. Are you happy
> with it?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/OpenAstronomy/openastronomy.github.io/pull/34/files#r53376520>
> .
>